contracts/CreditGuild.sol

```diff
@@ -0,0 +1,182 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.4;
+
+import "hardhat/console.sol";
+import "./ERC721Enumerable.sol";
+// import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@unioncredit/v1-sdk/contracts/BaseUnionMember.sol";
+
+/*
+credit guild
+*/
+
+// TODO: ask gerald about making nft non transferrable
+contract CreditGuild is ERC721Enumerable, BaseUnionMember, Ownable { 
+
+    bool public isInitialized;
+
+    uint256 public constant MIN_MEMBERSHIP_FEE = 1 ether;
+
+    uint256 public id;
+
+    uint256 public membershipFee;
+    
+    uint256 public vouchAmount;
+
+    event Initialize(address[] initialMembers);
+    event SetVouchAmount(uint256 vouchAmount);
+    event Register(address sender, address[] members, uint256 vouchAmount);
+    event SetMembershipFee(uint256 membershipFee);
+    event ClaimVouch(address sender, uint256 vouchAmount);
+    event MintNFT(address member, uint256 currentId);
+    event BurnMembership(address member, uint256 memberId);
+    event CheckIsMember(bool result);
+    event Stake(uint256 daoBalance);
+
+    constructor(
+      uint256 _membershipFee,
+      uint256 _vouchAmout,
+      address _marketRegistry,
+      address _unionToken,
+      address _underlyingToken
+    ) ERC721("UnionNFT", "UNFT") BaseUnionMember(_marketRegistry, _unionToken, _underlyingToken) {
+        
+        // set membership fee
+        require(_membershipFee >= MIN_MEMBERSHIP_FEE, "membershipFee too low");
+        membershipFee = _membershipFee;
+        vouchAmount = _vouchAmout;
+        isInitialized = false;
+        
+    }
+    // nouns dao v1 logic, nouns dao immutable, ercenumerable
+
+    // Seed the DAO with 3 members
+    function initialize(
+        address member1, 
+        address member2, 
+        address member3
+    ) public onlyOwner {
+        
+        require(!isInitialized, "initialized");
+        isInitialized = true;
+
+        // make sure all addresses are unique
+        require(member1 != member2, "!unique");
+        require(member2 != member3, "!unique");
+        require(member1 != member3, "!unique");
+
+        address[] memory initialMembers = new address[](3);
+        initialMembers[0] = member1;
+        initialMembers[1] = member2;
+        initialMembers[2] = member3;
+
+        require(userManager.checkIsMember(address(this)), "DAO !member");
+
+        for (uint256 i = 0; i < 3; i++) {
+            address member = initialMembers[i];
+
+            // check that this member is vouching for the DAO
+            // TODO: shouldn't this be: getVouchingAmount(address(this), member) ?
```

Comment:
Here you want to check that the member is vouching for the contract. The signature is `getVouchingAmount(address staker, address borrower)`  the staker is the member and the borrower is the contract (`address(this)`)